javalib/src/main/scala/java/math/BigDecimal.scala

```diff
@@ -0,0 +1,1761 @@
+   *
+   *  @param parityBit can be 0 or 1, it's only used in the case {@code HALF_EVEN}
+   *  @param fraction the mantissa to be analyzed
+   *  @param roundingMode the type of rounding
+   *  @return the carry propagated after rounding.
+   */
+  private def roundingBehavior(parityBit: Int, fraction: Int,
+      roundingMode: RoundingMode): Int = {
+    import RoundingMode._
+
+    val absFraction = Math.abs(fraction)
+    val sigFraction = java.lang.Integer.signum(fraction)
+    // the carry after rounding
+    roundingMode match {
+      case UP          => sigFraction
+      case DOWN        => 0
+      case CEILING     => Math.max(sigFraction, 0)
+      case FLOOR       => Math.min(sigFraction, 0)
+      case HALF_UP     => if (absFraction >= 5) sigFraction else 0
+      case HALF_DOWN   => if (absFraction > 5) sigFraction else 0
+      case HALF_EVEN   => if (absFraction + parityBit > 5) sigFraction else 0
+      case UNNECESSARY =>
+        if (fraction == 0) 0
+        else throw new ArithmeticException("Rounding necessary")
+    }
+  }
+
+  private def safeLongToInt(longValue: Long): Int = {
+    if (longValue < Int.MinValue || longValue > Int.MaxValue)
+      throw new ArithmeticException("Out of int range: " + longValue)
+
+    longValue.toInt
+  }
+
+  /** The value 0 with the most approximated scale of type {@code int}.
+   *
+   *  If {@code longScale > Integer.MAX_VALUE} the scale will be
+   *  {@code Integer.MAX_VALUE}.
+   *
+   *  If {@code longScale < Integer.MIN_VALUE} the scale will be
+   *  {@code Integer.MIN_VALUE}.
+   *
+   *  Otherwise {@code longScale} is casted to the type {@code int}.
+   *
+   *  @param longScale the scale to which the value 0 will be scaled.
+   *  @return the value 0 scaled by the closer scale of type {@code int}.
+   *  @see #scale
+   */
+  private def zeroScaledBy(longScale: Long): BigDecimal = {
+    if (longScale == longScale.toInt)
+      valueOf(0, longScale.toInt)
+    else if (longScale >= 0)
+      new BigDecimal(0, Int.MaxValue)
+    else
+      new BigDecimal(0, Int.MinValue)
+  }
+
+  protected def bitLength(sValue: Long): Int = {
+    val smallValue = if (sValue < 0) ~sValue else sValue
+    64 - java.lang.Long.numberOfLeadingZeros(smallValue)
+  }
+
+  private def bitLength(sValue: Int): Int = {
+    val smallValue = if (sValue < 0) ~sValue else sValue
+    32 - java.lang.Integer.numberOfLeadingZeros(smallValue)
+  }
+
+  @inline
+  private def charNotEqualTo(c: Char, cs: Char*): Boolean = !cs.contains(c)
+
+  @inline
+  private def charEqualTo(c: Char, cs: Char*): Boolean = cs.contains(c)
+
+  @inline
+  private def insertString(s: String, pos: Int, s2: String): String =
+    s.substring(0, pos) + s2 + s.substring(pos)
+
+  @inline
+  private def insertString(s: String, pos: Int, s2: String, s2Start: Int,
+      s2Len: Int): String =
+    insertString(s, pos, s2.substring(s2Start, s2Start + s2Len))
```

Comment:
Because the signature does not fit on one line and has to be wrapped, you should also use `{}` to enclose the body of the method, even if it is a one-liner. Otherwise it looks odd.
```scala
  private def insertString(s: String, pos: Int, s2: String, s2Start: Int,
      s2Len: Int): String = {
    insertString(s, pos, s2.substring(s2Start, s2Start + s2Len))
  }
```